src/EditorFeatures/Core.Wpf/InlineParamNameHints/InlineParamNameHintsTagger.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+
+namespace Microsoft.CodeAnalysis.Editor.InlineParamNameHints
+{
+    /// <summary>
+    /// The purpose of this tagger is to convert the InlineParamNameHintsDataTags to
+    /// the InlineParamNameHintsTag, which actually creates the UIElement. It reacts to
+    /// tags changing and updates the adornments accordingly.
+    /// </summary>
+    class InlineParamNameHintsTagger : ITagger<IntraTextAdornmentTag>
```

Comment:
should this implement IAccurateTAgger as well @jasonmalinowski ?

Reply:
would liek this to derive from FOregroundThreadAffinitizedObject and be updated to have threading asserts.